File: server/src/main/java/com/pass/exam/domain/ExamTitle.java
package com.pass.exam.domain;

import com.pass.global.exception.BadRequestException;
import jakarta.persistence.Column;
import jakarta.persistence.Embeddable;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Embeddable
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Getter
public class ExamTitle {

    private static final int MAX_TITLE_LENGTH = 100;

    @Column(name = "title", nullable = false)
    private String value;

    public ExamTitle(String value) {
        validateTitleLength(value);

        this.value = value;
    }

    private void validateTitleLength(String value) {
        if (value == null || value.isBlank()) {
            throw new BadRequestException("시험 제목은 비어있을 수 없습니다.");
        }

        if (value.length() > MAX_TITLE_LENGTH) {
            throw new BadRequestException("시험 제목은 %d자 이하여야 합니다.".formatted(MAX_TITLE_LENGTH));
        }
    }
}

File: server/src/main/kotlin/com/fluffy/exam/domain/ExamTitle.kt
package com.fluffy.exam.domain

import com.fluffy.global.exception.BadRequestException
import jakarta.persistence.Column
import jakarta.persistence.Embeddable

@Embeddable
data class ExamTitle(

    @field:Column(name = "title", nullable = false)
    val value: String
) {

    companion object {

        private const val MAX_TITLE_LENGTH = 100
    }

    init {
        validateTitleNotBlank()
        validateTitleLength()
    }

    private fun validateTitleNotBlank() {
        if (value.isBlank()) {
            throw BadRequestException("시험 제목은 비어있을 수 없습니다.")
        }
    }

    private fun validateTitleLength() {
        if (value.length > MAX_TITLE_LENGTH) {
            throw BadRequestException("시험 제목은 ${MAX_TITLE_LENGTH}자 이하여야 합니다.")
        }
    }
}